### PR TITLE
Dynamic consumer

### LINKED ISSUE #6: Creating a dynamic group.id for kafka topic using HOSTNAME variable in the container
While deploying the service i want it have different consumer group.id for different instance as vectors are cached separately on instances. This will make it easy during deployment to reset the respective consumer offset

### PR DESCRIPTION
**Pull request by MirMustafaAli**
Fixes #6 [https://github.com/CoffeeBeansLabs/SimDB/issues/6#issue-749343538](url). 
- A new field in app config **mode**--> **consumer_group_mode** as inner key.
- method added to get consumer mode from app-config
- method to get consumer id which shall be replace _group.id_ in app-config
@karthikcb @indraneelr

### CODE REVIEW

File: settings/config.py [hunk 1]
```diff
@@ -86,3 +86,6 @@ def get_reader_fetch_freq(self):
 
   def get_indexer_retrain_freq(self):
     return self.config["indexer"]["retrain_frequency"]
+
+  def get_consumer_group_mode(self):
+    return self.config["mode"]["consumer_group_mode"]
```
Comment (indraneelr) [inline settings/config.py:91]: where will this be in `app-config.json`?  this should be in connectors.readers["kafka_reader"] block.

File: settings/factory.py [hunk 2]
```diff
@@ -69,6 +71,8 @@ def get_reader(self):
 
     if reader_conf["name"] == 'kafka_reader':
       print("creating Kafka reader..")
+      reader_conf["connection"]["settings"]["group.id"] = self.get_consumer_group_id(reader_conf)
```
Comment (indraneelr) [inline settings/factory.py:74]: this should also support the default method of picking consumer group directly from the app-config.json

**Review by indraneelr — CHANGES_REQUESTED** [review]

File: settings/config.py (hunk 1 above)
Comment (MirMustafaAli) [reply settings/config.py:91]: right now i have added an extra field named _"mode"_. and the format is shown below

>  "mode": {
>     "consumer_group_mode": "hostname"
>   }

File: settings/factory.py (hunk 2 above)
Comment (MirMustafaAli) [reply settings/factory.py:74]: - yes, if someone wants default method they must not add that mode field

File: settings/config.py [hunk 3]
```diff
@@ -87,5 +87,4 @@ def get_reader_fetch_freq(self):
   def get_indexer_retrain_freq(self):
     return self.config["indexer"]["retrain_frequency"]
 
-  def get_consumer_group_mode(self):
-    return self.config["mode"]["consumer_group_mode"]
```
Comment (MirMustafaAli) [inline settings/config.py]: removed the method to check consumer group mode

File: settings/factory.py [hunk 4]
```diff
@@ -127,7 +127,7 @@ def get_writer(self):
 
   def get_consumer_group_id(self, reader_conf):
     try:
-      if self.config.get_consumer_group_mode() == "hostname":
```
Comment (MirMustafaAli) [inline settings/factory.py]: using _reader_conf_ to access _consumer_group_mode_

File: settings/factory.py [hunk 5]
```diff
@@ -120,3 +124,16 @@ def get_writer(self):
       writer = RedisWriter(writer_conf, mapper, rank_field)
 
     return writer
+
+  def get_consumer_group_id(self, reader_conf):
+    try:
+      if self.config.get_consumer_group_mode() == "hostname":
+        # todo: need to log the exception
```
Comment (indraneelr) [inline settings/factory.py:131]: you need to return the consumer group name from here right?

Comment (MirMustafaAli) [reply settings/factory.py:131]: yes, i will returning from here

**Pull request merged**